File: project-euler/Python/problem2.py
#!/usr/bin/env python
"""
Python solution for Project Euler Problem 2: Even Fibonacci numbers
"""
__author__ = "Emelie Widegren"
__email__ = "emeliewide@gmail.com"


def even_fibonacci():
    """
    Returns the sum of the even-valued terms below four million.
    """
    a, b = 1, 1
    sum = 0
    while b < 4000000:
        a, b = b, a + b
        if b % 2 == 0:
            sum += b
    return sum


if __name__ == '__main__':
    print(even_fibonacci())

File: project-euler/Java/Problem2.java
import java.util.ArrayList;

/**
 * Java solution for Project Euler Problem 2: Even Fibonacci numbers
 *
 * @author Emelie Widegren <widegren.emelie@gmail.com>
 */
public class Problem2 {

  public static void main(String arg[]) {
    evenFibonacci();
  }

  /** Returns the sum of the even-valued terms below four million. */
  private static void evenFibonacci() {
    ArrayList<Integer> list = new ArrayList<Integer>();
    int result = 0, i = 0, fib = 1;

    while (fib < 4000000) {
      list.add(fib);
      if (i != 0) {
        fib = fib + list.get(i - 1);
        if (fib % 2 == 0) {
          result = result + fib;
        }
      }
      i++;
    }
    System.out.println(result);
  }
}
